src/util/config.rs

```diff
@@ -0,0 +1,178 @@
+// Copyright 2016 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::error;
+use std::fmt;
+use rocksdb::DBCompressionType;
+
+#[derive(Debug)]
+pub struct ParseConfigError;
+
+impl fmt::Display for ParseConfigError {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        write!(fmt, "malformed property")
+    }
+}
+
+impl error::Error for ParseConfigError {
+    fn description(&self) -> &str {
+        "malformed property"
+    }
+}
+
+pub fn parse_rocksdb_compression(tp: &str) -> Result<DBCompressionType, ParseConfigError> {
+    match &*tp.to_owned().to_lowercase() {
+        "no" => Ok(DBCompressionType::DBNo),
+        "snappy" => Ok(DBCompressionType::DBSnappy),
+        "zlib" => Ok(DBCompressionType::DBZlib),
+        "bzip2" => Ok(DBCompressionType::DBBz2),
+        "lz4" => Ok(DBCompressionType::DBLz4),
+        "lz4hc" => Ok(DBCompressionType::DBLz4hc),
+        _ => Err(ParseConfigError),
+    }
+}
+
+fn split_property(property: &str) -> Result<(f64, &str), ParseConfigError> {
+    let mut indx = 0;
+    for s in property.chars() {
+        match s {
+            '0'...'9' | '.' => {
+                indx += 1;
+            }
+            _ => {
+                break;
+            }
+        }
+    }
+
+    let (num, unit) = property.split_at(indx);
+    num.parse::<f64>().map(|f| (f, unit)).or(Err(ParseConfigError))
+}
+
+const UNIT: i64 = 1;
+const DATA_MAGNITUDE: i64 = 1024;
+const KB: i64 = UNIT * DATA_MAGNITUDE;
+const MB: i64 = KB * DATA_MAGNITUDE;
+const GB: i64 = MB * DATA_MAGNITUDE;
+// When we cast TB and PB to `f64`, compiler will report a overflow warning.
+// TB = GB * DATA_MAGNITUDE
+// PB = PB * DATA_MAGNITUDE
+
+const TIME_MAGNITUDE_1: i64 = 1000;
+const TIME_MAGNITUDE_2: i64 = 60;
+const MS: i64 = UNIT;
+const SECOND: i64 = MS * TIME_MAGNITUDE_1;
+const MINTUE: i64 = SECOND * TIME_MAGNITUDE_2;
+const HOUR: i64 = MINTUE * TIME_MAGNITUDE_2;
+
+pub fn parse_readable_int(size: &str) -> Result<i64, ParseConfigError> {
+    let (num, unit) = try!(split_property(size));
+
+    match &*unit.to_owned().to_lowercase() {
+        // file size
+        "kb" => Ok((num * (KB as f64)) as i64),
+        "mb" => Ok((num * (MB as f64)) as i64),
+        "gb" => Ok((num * (GB as f64)) as i64),
+        "tb" => {
+            let mut ret: i64 = 0;
```

Comment:
All clear.
After changing type to `uszie`, all overflow warning is gone, and no test fail. Awesome @BusyJay